inventory/any.cf

```diff
@@ -520,38 +520,30 @@ bundle agent cfe_autorun_inventory_aws_ec2_metadata
 {
   methods:
     !(disable_inventory_aws|disable_inventory_aws_ec2_metadata)::
-      "cfe_autorun_inventory_aws_ec2_metadata_cache";
+      "cfe_autorun_inventory_aws_ec2_metadata_cache"
+        action => if_elapsed_day;
       "cfe_aws_ec2_metadata_from_cache";
 }
 bundle agent cfe_autorun_inventory_aws_ec2_metadata_cache
-# @brief Cache ec2 metadata from http request for one day
+# @brief Cache ec2 metadata from http request
+# It's up to caller to ensure this bundle is called only once a day
 #
 # Provides cache of ec2 instance metadata for inventory
 {
 
   vars:
-
     ec2_instance.!(disable_inventory_aws|disable_inventory_aws_ec2_metadata)::
-
-      "curl"      string => "$(paths.curl)";
-      "curl_opts" string => "--max-time 1 --connect-timeout 1 --silent";
       "URL"       string => "http://169.254.169.254/latest/dynamic/instance-identity/document";
       "cache"     string => "$(sys.statedir)/aws_ec2_metadata";
-
-  commands:
-
-    ec2_instance.!(disable_inventory_aws|disable_inventory_aws_ec2_metadata)::
-
-      "$(curl)"
-        args => "$(curl_opts) $(URL) -o $(cache)",
-        if => and( not( fileexists( $(cache) )),
-                   isexecutable( $(curl) ));
+      "response"  data => url_get($(URL), '{"url.max_content": 512000, "url.timeout": 1}');
```

Comment:
If you want to delay url_get being called this to when the bundle is actually called, it will need to depend on some other promises (like a files promise that is only evaluated when the bundle is called).

Reply:
Perhaps you can place the action body directly on this promise so that it only evaluates once a day.

Reply:
nope. It's not mentioned in [ifelapsed docs][ie], but is said [elsewhere][ew] that "vars promises don't support promise locking". Also CFE-3162 is somewhat relevant.


[ie]: https://docs.cfengine.com/docs/3.12/reference-promise-types.html#ifelapsed 
[ew]: https://docs.cfengine.com/docs/3.12/reference-language-concepts-promises.html#promise-locking
[CFE-3162]: https://tracker.mender.io/browse/CFE-3162

Reply:
so I'll make it depend on some dummy promise, then.